### ISSUE TITLE
[9.1.2] ticket update - add a second requester get a invalid requester entry

### ISSUE THREAD

**Issue by Armin83** [glpi-project/glpi#1806]
I have a already created ticket and add a addional requester.
After i add the second requester i have three requester. Two normal requester, that i know and spect, and one blank / invalid requester.

In Debug mode there is a small error but disappear too fast.

![1](https://cloud.githubusercontent.com/assets/19192087/22824575/c65d0bac-ef88-11e6-8fa6-e2bc4608a4a6.jpg)

![2](https://cloud.githubusercontent.com/assets/19192087/22824576/c89d8860-ef88-11e6-98b8-3af6e952b4d3.jpg)
![3](https://cloud.githubusercontent.com/assets/19192087/22824577/ca36459a-ef88-11e6-8e9a-6d88b430041e.jpg)

9.1.1 all ok
91.2 get this bug

**Comment by Blank-Vcf** [comment]
I have the same problem on 9.1.2. If i look in db i have only 3 users ( 2 request and 1 assigned)

**Comment by veerlegeorges** [comment]
We have the samen problem on 9.1.2.
Error in debug mode is :
PHP Notice : Array to string conversion in inc/commondbtm.class.php at line 426
PHP Warning : strlen() expects parameter 1 to be string, array given in inc/commondbtm.class.php at line 158

**Comment by ladenree76000** [comment]
confirmed on 9.1.2

**Comment by tsmr** [comment]
Correction : 

replace into inc/commonitilobject.class.php function addAdditionalActors

l1655
```
if (isset($tmp['users_id'])) {
```
by
```
if (isset($tmp['users_id']) && !is_array($tmp['users_id'])) {
```

**Comment by veerlegeorges** [comment]
Works ! Thx ! :-)

**Comment by trasher** [comment]
I'm not able to reproduce with current 9.1/bugfixes.

@tomolimo said that he reproduce with a specific rule active (see https://github.com/glpi-project/glpi/issues/2030#issuecomment-296591073); but even with a rule, I do not have the problem.

Maybe the problem have been solved, or maybe I miss informations on how to reproduce. Could someone test with 9.1/bugfixes?

By the way; I do not really agree with the proposed fix (even if it seems to solve the issue); since the issue may be the same elsewhere in the code. I guess the real reason is not here.

**Comment by tomolimo** [comment]
Hello,
I'll try to reproduce with the rule, and I'll keep you informed,
regards,
Tomolimo

**Comment by trasher** [comment]
Great; thank you very much :)

**Comment by tomolimo** [comment]
I confirm that if I re-activate the rule, then I get this:
![image](https://cloud.githubusercontent.com/assets/7569958/25343621/e24f4f58-290f-11e7-949f-aab3f6ef1c6d.png)
each time I try to add a requester to a ticket (even if the rule doesn't apply to this ticket)

**Comment by tomolimo** [comment]
The rule is very simple:

Let's imagine an entity tree:
root > B > C
And you want to automatically add SLA (or SLT: TTR and TTO) to any tickets under entity B

that's all,

**Comment by tomolimo** [comment]
To give complete path to reproduce:

1. entity tree is:
Root entity > Global > Local IT > INDIA

1. define a calendar:
![image](https://cloud.githubusercontent.com/assets/7569958/25374962/8b5c4864-299f-11e7-9150-83091c83b571.png)
No close time

1. define SLA:
![image](https://cloud.githubusercontent.com/assets/7569958/25375018/b6d83412-299f-11e7-97a3-e19e3cbedda4.png)

1. define SLT: (TTO and TTR)
![image](https://cloud.githubusercontent.com/assets/7569958/25375113/f909f618-299f-11e7-9d31-766d36544f3a.png)
With no escalation levels

1. define a rule that will apply on tickets in India entity:
Criteria:
![image](https://cloud.githubusercontent.com/assets/7569958/25375334/ac0bc2aa-29a0-11e7-99b6-96c65160c5fc.png)
and
Actions:
![image](https://cloud.githubusercontent.com/assets/7569958/25375383/ca0d20a0-29a0-11e7-9d80-a5c25ab94da3.png)

Then, any ticket to which you want to add a new requester, will get the ghost actor (id=0) added to it, and whatever the ticket entity is!

If this rule is de-activated, then there is no problem with the requesters of the tickets

Regards,
Tomolimo

**Comment by trasher** [comment]
Hello,

Thanks for the details. I'll try to reproduce on my side today.

**Comment by trasher** [comment]
I've just tested... And still can't reproduce :/

**Comment by tomolimo** [comment]
it's 9.1.2

**Comment by trasher** [comment]
Inded, I can reproduce on 9.1.2. SO this is fixed in current 9.1/bugfixes.
